File: Java/Gradle/algorithms/src/main/java/org/algorithms/example/redblacktree/Color.java
package org.algorithms.example.redblacktree;

/**
 * @author korov
 */
public enum Color {
    /**
     * 红色
     */
    RED(0),

    /**
     * 黑色
     */
    BLACK(1);

    private final Integer value;


    Color(int value) {
        this.value = value;
    }

    public Integer getValue() {
        return value;
    }
}

File: Java/Gradle/bdd/src/main/kotlin/org/korov/kotlin/Color.kt
package org.korov.kotlin

/**
 * 最简单的枚举类
 */
enum class Color {
    RED, YELLOW, WHITE
}

/**
 * 带属性的枚举类
 */
enum class Color1(val r: Int, val g: Int, val b: Int) {
    RED(255, 0, 0),
    YELLOW(255, 255, 0);

    fun rgb() = (r * 256 + g) * 256 + b
}

fun main() {
    println(Color1.YELLOW.rgb())
    var color = Color.YELLOW
    when (color) {
        Color.WHITE -> println("White")
        Color.RED, Color.YELLOW -> println("not White")
    }
}


